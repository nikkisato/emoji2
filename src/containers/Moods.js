import React, { useReducer } from 'react';
import Controls from '../components/controls/Controls';
import Face from '../components/face/Face';
import {
  drinkCoffee,
  eatSnacks,
  takeNaps,
  study
} from '../actions/moodActions';
import { getFace } from '../selectors/moodSelectors';
import reducer from '../reducers/moodReducers';

const actions = [
  { name: 'DRINK_COFFEE', text: 'Drink Coffee', stateName: 'coffees' },
  { name: 'EAT_SNACK', text: 'Snack', stateName: 'snacks' },
  { name: 'TAKE_NAP', text: 'Nap', stateName: 'naps' },
  { name: 'STUDY', text: 'Study', stateName: 'studies' }
];
export default function Moods() {
  const [state, dispatch] = useReducer(reducer, {
    coffees: 0,
    naps: 0,
    studies: 0,
    snacks: 0
  });
  const face = getFace(state);

  return (
    <>
      <Controls actions={actions} dispatch={dispatch} />
      <Face emoji={face} />
    </>
  );
}
