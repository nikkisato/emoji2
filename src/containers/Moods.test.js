import React from 'react';
import { shallow } from 'enzyme';
import Moods, {
  isTired,
  isHyper,
  isEducated,
  isHungry,
  getFace
} from './Moods';

describe('Moods', () => {
  describe('container', () => {
    it('matches a snapshot', () => {
      const wrapper = shallow(<Moods />);
      expect(wrapper).toMatchSnapshot();
    });
  });
});
