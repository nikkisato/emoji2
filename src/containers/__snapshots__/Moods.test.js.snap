// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Moods container matches a snapshot 1`] = `
<Fragment>
  <Controls
    actions={
      Array [
        Object {
          "name": "DRINK_COFFEE",
          "stateName": "coffees",
          "text": "Drink Coffee",
        },
        Object {
          "name": "EAT_SNACK",
          "stateName": "snacks",
          "text": "Snack",
        },
        Object {
          "name": "TAKE_NAP",
          "stateName": "naps",
          "text": "Nap",
        },
        Object {
          "name": "STUDY",
          "stateName": "studies",
          "text": "Study",
        },
      ]
    }
    dispatch={[Function]}
  />
  <Face
    emoji="ðŸ¤¬"
  />
</Fragment>
`;
